extensions/interactions/NumericExpressionInput/directives/oppia-interactive-numeric-expression-input.component.ts

```diff
@@ -20,108 +20,136 @@
-        });
+  isCurrentAnswerValid(): boolean {
+    let activeGuppyObject = (
+      this.guppyInitializationService.findActiveGuppyObject());
+    if (this.hasBeenTouched && activeGuppyObject === undefined) {
+      // Replacing abs symbol, '|x|', with text, 'abs(x)' since the symbol
+      // is not compatible with nerdamer or with the backend validations.
+      this.value = this.mathInteractionsService.replaceAbsSymbolWithText(
+        this.value);
+      let answerIsValid = (
+        this.mathInteractionsService.validateNumericExpression(
+          this.value));
+      if (answerIsValid) {
+        // Explicitly inserting '*' signs wherever necessary.
+        this.value = this.mathInteractionsService.insertMultiplicationSigns(
+          this.value);
+      }
+      this.warningText = this.mathInteractionsService.getWarningText();
+      return answerIsValid;
+    }
+    this.warningText = '';
+    return true;
+  }
 
-        CurrentInteractionService.registerCurrentInteraction(
-          ctrl.submitAnswer, ctrl.isCurrentAnswerValid);
-      };
+  submitAnswer(): void {
+    if (!this.isCurrentAnswerValid()) {
+      return;
     }
-  ]
-});
+    this.currentInteractionService.onSubmit(this.value, (
+      this.numericExpressionInputRulesService
+    ) as unknown as InteractionRulesService);
+  }
+
+  showOSK(): void {
+    this.guppyInitializationService.setShowOSK(true);
+    GuppyInitializationService.interactionType = 'NumericExpressionInput';
+  }
+
+  ngOnInit(): void {
+    this.hasBeenTouched = false;
+    this.guppyConfigurationService.init();
+    const { useFractionForDivision, placeholder } = (
+      this.interactionAttributesExtractorService.getValuesFromAttributes(
+        'NumericExpressionInput', this._getAttributes()
+      )) as NumericExpressionInputCustomizationArgs;
+    this.guppyConfigurationService.changeDivSymbol(
+      (useFractionForDivision as unknown as { value: boolean }).value);
+    this.guppyInitializationService.init(
+      'guppy-div-learner',
+      placeholder.value.unicode,
+      this.savedSolution !== undefined ?
+      this.savedSolution as string : ''
+    );
+    let eventType = (
+      this.deviceInfoService.isMobileUserAgent() &&
+      this.deviceInfoService.hasTouchEvents()) ? 'focus' : 'change';
+    // We need the 'focus' event while using the on screen keyboard (only
+    // for touch-based devices) to capture input from user and the 'change'
+    // event while using the normal keyboard.
+    Guppy.event(eventType, () => {
+      let activeGuppyObject = (
+        this.guppyInitializationService.findActiveGuppyObject());
+      if (activeGuppyObject !== undefined) {
+        this.hasBeenTouched = true;
+        this.value = activeGuppyObject.guppyInstance.asciimath();
+        if (eventType === 'change') {
+          // Need to manually trigger the digest cycle to make any
+          // 'watchers' aware of changes in answer.
+          // $rootScope.$applyAsync();
+        }
```

Comment:
Why is this `if` needed when everything is commented?

Reply:
done!